spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/batch/JdbcSearchableJobExecutionDao.java

```diff
@@ -639,16 +643,42 @@ JobExecution createJobExecutionFromResultSet(ResultSet rs, int rowNum) throws SQ
 		jobExecution = new JobExecution(jobInstance, jobParameters);
 		jobExecution.setId(id);
 
-		jobExecution.setStartTime(rs.getObject(2, LocalDateTime.class));
-		jobExecution.setEndTime(rs.getObject(3, LocalDateTime.class));
+		jobExecution.setStartTime(getLocalDateTime(2, rs));
+		jobExecution.setEndTime(getLocalDateTime(3, rs));
 		jobExecution.setStatus(BatchStatus.valueOf(rs.getString(4)));
 		jobExecution.setExitStatus(new ExitStatus(rs.getString(5), rs.getString(6)));
-		jobExecution.setCreateTime(rs.getObject(7, LocalDateTime.class));
-		jobExecution.setLastUpdated(rs.getObject(8, LocalDateTime.class));
+		jobExecution.setCreateTime(getLocalDateTime(7, rs));
+		jobExecution.setLastUpdated(getLocalDateTime(8, rs));
 		jobExecution.setVersion(rs.getInt(9));
 		return jobExecution;
 	}
 
+	private  LocalDateTime getLocalDateTime(int columnIndex, ResultSet rs) throws SQLException {
+		LocalDateTime result = null;
+		if (databaseType == DatabaseType.DB2) {
+			try {
+				result = rs.getObject(columnIndex, LocalDateTime.class);
+			} catch (NullPointerException npe) {
+				if (!npe.getMessage().contains("java.sql.Timestamp.toLocalDateTime()\" because \"<local4>\" is null")) {

```

Comment:
Since we bypass an NPE which is a bug, let's make a note as TODO when this eventually gets fixed at DB2. Also, it would probably be a better to log a message at the else block stating the actual exception we are bypassing.